common/src/utils/io/logging.rs

```diff
@@ -604,11 +604,48 @@ impl Logger {
     }
 }
 
+/// Set `RUST_LOG` variable to env if does not exist
+///
+/// ```
+/// use heimdall_common::utils::io::logging::set_logger_env;
+///
+/// let verbosity = clap_verbosity_flag::Verbosity::new(-1, 0);
+/// set_logger_env(&verbosity);
+/// ```
+pub fn set_logger_env(verbosity: &clap_verbosity_flag::Verbosity) {
+    let env_not_set = std::env::var("RUST_LOG").is_err();
+
+    if env_not_set {
+        let log_level = match verbosity.log_level() {
+            Some(level) => level.as_str(),
+            None => "SILENT",
+        };
+
+        std::env::set_var("RUST_LOG", log_level);
+    }
+}
+
+/// Returns a new instance of `Logger` and `TraceFactory` given a log level
+///
+/// ```
+/// use heimdall_common::utils::io::logging::get_logger_and_trace;
+///
+/// let verbosity = clap_verbosity_flag::Verbosity::new(-1, 0);
+/// get_logger_and_trace(&verbosity);
+/// ```
+pub fn get_logger_and_trace(verbosity: &clap_verbosity_flag::Verbosity) -> (Logger, TraceFactory) {
```

Comment:
for now you can actually just use `Logger::new`, it'll return `(Logger, TraceFactory)`

I do have #126 open which will update this functionality eventually (all logger methods being moved to macros, `TraceFactory` separated from `Logger`.